src/components/ChatEntry.jsx

```diff
@@ -1,20 +1,39 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
 import './ChatEntry.css';
+import TimeStamp from './TimeStamp';
+
+const ChatEntry = ({sender, body, timeStamp, onLike, onUnlike}) => {
+  const [buttonText, setButtonText] = useState('🤍');
```

Comment:
We shouldn't need to use `state` to keep track of the heart because we can get `liked` from each message. We should compute whether the heart is red or white by looking at messages `liked` value. We should prefer to not introduce state, which introduces additional complexity into a project, if we have some other way to accomplish the feature.